### PR TITLE
F/eslint cleanup

### PR DESCRIPTION
**Pull request by mattsbennett**
Just a few simple cleanup changes to get eslint working/cleaned up

- [x] Fix eslint config (jest/globals env config seems to require extending jest/recommended)
- [x] Fix eslint errors project-wide (most auto-fixed, some required manual fixes in documentStateTest.ts)

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -2,11 +2,12 @@ module.exports = {
     env: {
         browser: true,
         es2021: true,
-        "jest/globals": true,
+        'jest/globals': true,
     },
     extends: [
         'airbnb-base',
         'plugin:import/typescript',
+        'plugin:jest/recommended',
```
Comment (glenreesor) [inline .eslintrc.js:10]: Ah, this explains why I had an error before (which I promptly forgot about :smile: )

File: src/components/DisplayedDocument.ts [hunk 2]
```diff
@@ -199,9 +199,7 @@ function DisplayedDocument(): m.Component<Attrs> {
 
             childIds.forEach((childId) => {
                 calculateDimensions(allDimensions, childId);
-                totalChildrenHeight += safeGetDimensions(
-                    allDimensions, childId,
-                ).heightWithChildren;
+                totalChildrenHeight += safeGetDimensions(allDimensions, childId).heightWithChildren;
```
Comment (glenreesor) [inline src/components/DisplayedDocument.ts:202]: I'm curious -- was this an editor auto format thing, or a personal preference re: line length?

File: src/components/DisplayedDocument.ts [hunk 3]
```diff
@@ -275,7 +273,6 @@ function DisplayedDocument(): m.Component<Attrs> {
      * will be longer than our max node width
      *
      * @param contents The contents of the node to be rendered
-     *
```
Comment (glenreesor) [inline src/components/DisplayedDocument.ts:278]: Another thing I'm curious about, as this didn't trigger a linting error for me.

File: src/state/documentState.test.ts [hunk 4]
```diff
@@ -226,30 +225,30 @@ describe('deleteNode', () => {
...
+            docState.getNodeContents(nodeToDeleteId),
         ).toBe(nodeToDeleteContentsSrc);
     });
 });
 
 describe('getChildrenVisible', () => {
     it('returns true when there are no children', () => {
         const rootNodeId = docState.getRootNodeId();
-        expect(docState.getChildrenVisible(rootNodeId));
+        expect(docState.getChildrenVisible(rootNodeId)).toBe(true);
```
Comment (glenreesor) [inline src/state/documentState.test.ts:236]: Nice :+1:

File: src/state/documentState.test.ts [hunk 5]
```diff
@@ -467,13 +465,15 @@ describe('getRootNodeId', () => {
 });
 
 describe('getSelectedNodeId / setSelectedNodeId', () => {
+    it('gets/sets nodes', () => {
```
Comment (glenreesor) [inline src/state/documentState.test.ts:468]: Ha! Good catch!

File: src/state/documentState.test.ts [hunk 6]
```diff
@@ -656,29 +656,23 @@ describe('redo / undo / getRedoIsAvailable / getUndoIsAvailable', () => {
 
         const reversedMapsBeforeChange = [...serializedMapsBeforeChange].reverse();
         reversedMapsBeforeChange.forEach((serializedMap, index) => {
+            const expectedUndoIsAvailable = (index !== serializedMapsBeforeChange.length - 1);
```
Comment (glenreesor) [inline src/state/documentState.test.ts:659]: Also nice!

**Review by glenreesor — COMMENTED** [review]
Hey @mattsbennett good stuff! Just a couple of clarifying questions. re: formatting, I'm toying with the idea of using prettier. It seems to have fixed some of the odd formatting we encountered when we previously investigated it (in particular ternaries, I think). I use it at work and it's really nice to not even have to think about it.

Anyway, just a consideration at this point.

File: src/components/DisplayedDocument.ts (hunk 2 above)
Comment (mattsbennett) [reply src/components/DisplayedDocument.ts:202]: Ah, yeah, this was just my editor enforcing the eslint rules configured in `.eslintrc.js`. Looks like this one came from `'airbnb-base'` (`function-paren-newline` rule).

File: src/components/DisplayedDocument.ts (hunk 3 above)
Comment (mattsbennett) [reply src/components/DisplayedDocument.ts:278]: Looks ilke this was another auto-fix - came from `'plugin:jsdoc/recommended'` (`jsdoc/tag-lines` rule). Undoing the fix, appears it only shows as a warning in VSCode (though `--fix` fixes it automatically):

Screen Shot 2022-02-03 at 8 33 16 PM

**Comment by mattsbennett** [comment]
Hey @glenreesor - thanks for taking a look - responses above 😃 I'm easy re: prettier - I've had some exposure to it and don't mind it (sometimes a bit jarring when it does things I wouldn't choose to do, but far more important things to spend time on, so I like that). Really simple to configure it alongside ESLint now as well.

File: src/components/DisplayedDocument.ts (hunk 2 above)
Comment (glenreesor) [reply src/components/DisplayedDocument.ts:202]: Ah, looks like my local eslint config was still broken. Fixed now, and I see that :+1:

File: src/components/DisplayedDocument.ts (hunk 3 above)
Comment (glenreesor) [reply src/components/DisplayedDocument.ts:278]: Also visible now that my eslint is fixed :stuck_out_tongue: .

**Comment by glenreesor** [comment]
> Hey @glenreesor - thanks for taking a look - responses above smiley I'm easy re: prettier - I've had some exposure to it and don't mind it (sometimes a bit jarring when it does things I wouldn't choose to do, but far more important things to spend time on, so I like that). Really simple to configure it alongside ESLint now as well.

Cool. I'm not in a huge rush since I'd like to get Cordova / Android going first, but thanks for your feedback!

**Pull request merged**